src/realm/util/interprocess_mutex.hpp

```diff
@@ -95,8 +95,18 @@ class InterprocessMutex {
     /// InterprocessMutex created on the same file (same inode on POSIX) share the same LockInfo.
     /// LockInfo will be saved in a static map as a weak ptr and use the UniqueID as the key.
     /// Operations on the map need to be protected by s_mutex
-    static std::map<File::UniqueID, std::weak_ptr<LockInfo>> s_info_map;
-    static Mutex s_mutex;
+    static std::map<File::UniqueID, std::weak_ptr<LockInfo>>* s_info_map;
+    static Mutex* s_mutex;
+    /// We control the order of initialization of these static variables,
+    /// creating them on the heap so that they last for the enitre lifetime
+    /// of the process. The destructor on these is never called, but the
+    /// process will clean up the memory when exiting. It would not be enough
+    /// to count instances of InterprocessMutex and clean up these statics when
+    /// the count reaches zero because the process could create more
+    /// InterprocessMutexes before the process ends, so we really need these
+    /// variables for the entire lifetime of the proces.
+    static std::once_flag static_flag;
+    static void static_initialisation();
```

Comment:
Being Canadian, I'm always torn between US and British spelling :( I'll try to be consistent with US going forward since it seems the rest of the code uses "initialize"